retesteth/RPCSession.cpp

```diff
@@ -264,30 +264,19 @@ string RPCSession::eth_getCode(string const& _address, string const& _blockNumbe
 
 test::scheme_block RPCSession::eth_getBlockByNumber(string const& _blockNumber, bool _fullObjects)
 {
-	// NOTE: to_string() converts bool to 0 or 1
-	Json::Value response = rpcCall("eth_getBlockByNumber", { quote(_blockNumber), _fullObjects ? "true" : "false" });
-    return test::scheme_block(dataobject::ConvertJsoncppToData(response));
+    return test::scheme_block(
+        rpcCall("eth_getBlockByNumber", {quote(_blockNumber), _fullObjects ? "true" : "false"}));
 }
 
 test::scheme_transactionReceipt RPCSession::eth_getTransactionReceipt(string const& _transactionHash)
 {
-	Json::Value const result = rpcCall("eth_getTransactionReceipt", { quote(_transactionHash) });
-    return test::scheme_transactionReceipt(dataobject::ConvertJsoncppToData(result));
+    return test::scheme_transactionReceipt(
+        rpcCall("eth_getTransactionReceipt", {quote(_transactionHash)}));
 }
 
-string RPCSession::eth_blockNumber()
+int RPCSession::eth_blockNumber()
 {
-    return rpcCall("eth_blockNumber", { }).asString();
-}
-
-string RPCSession::eth_sendTransaction(TransactionData const& _td)
-{
-	return rpcCall("eth_sendTransaction", { _td.toJson() }).asString();
-}
-
-string RPCSession::eth_call(TransactionData const& _td, string const& _blockNumber)
-{
-	return rpcCall("eth_call", { _td.toJson(), quote(_blockNumber) }).asString();
+    return rpcCall("eth_blockNumber", {}).asInt();
```

Comment:
Ethereum JSON-RPC standard is a hex encoded string (https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_blocknumber) - any reason this should vary from the standard?

Reply:
hm, looks like I get integer from geth

Reply:
The spec says string.  However it was coded before worked.